### ISSUE TITLE
It's great to see guys who love this old keyboard as well!☺

### ISSUE THREAD

**Issue by Curious-r** [vuckale/compaq-mx-11800-qmk-via#1]
I've also revamped this keyboard in the past few years because it's so beautifully laid out. But its PCB quality is so poor that it can be easily damaged during re-soldering, so I remaked the PCB and added some diodes for solving the ghost key.
![image](https://github.com/vuckale/compaq_mx_11800_qmk_via/assets/19733878/2f856663-da3d-4d01-87e1-cde11a1074a6)

**Comment by vuckale** [comment]
I agree this board is one of my favourites beacause of the layout. I liked it so much that I also remade the pcb for it including diodes, it's daughterboard and wrote firmware. Here is a quick pic

![pcb](https://github.com/vuckale/compaq_mx_11800_qmk_via/assets/56733276/44225b10-50c2-4d7d-92d3-883f53421f8a)

Check out the repository if you want: https://github.com/vuckale/va11800-pcb
Keep in mind that this is still work in progress. I have done 2 prototypes till now and it still has some alignment issues that need to be fixed.

**Comment by Curious-r** [comment]
@vuckale Cool! Mate, your PCB is so pretty. And I have a question, does your PCB with diodes work well with the original daughterboard? I lack testing about this.
And I am using "switch socket"(kailh etc.), so that we can implement switchs hot-swapping. But it is not done either.
 
![image](https://github.com/vuckale/compaq_mx_11800_qmk_via/assets/19733878/a1b997c9-84de-4cf6-bf61-ed7076015333)
![image](https://github.com/vuckale/compaq_mx_11800_qmk_via/assets/19733878/d9bea78d-371e-4583-8f44-ff26f1756480)

**Comment by vuckale** [comment]
@Curious-r Wow yours looks awesome too! Awesome idea to put Kailh sockets there. Are you going to design a switch plate too? 

My PCB does not work with original daughterboard. I have rewritten the matrix completely so I had to design a daughterboard too. I am using a teensy++ 2.0 to control the board, it has some cool features, for example you can use two rotary encoders, mini speaker, and solenoid, and multiple layout options. It works with QMK firmware and I will be porting it to VIA and VIAL when I have some time.

**Comment by Curious-r** [comment]
@vuckale Yes I have desgined a palte by `http://builder.swillkb.com/`, my layout is [here](http://www.keyboard-layout-editor.com/##@_backcolor=#222222&name=G80-11801-Crs&author=Curious&switchMount=cherry&switchBrand=cherry&switchType=MX1A-G1xx&pcb:true&plate:true;&@_c=#e3e2dd&t=#0a0808&p=DSA&sm=cherry&sb=cherry&st=MX1A-G1xx&a:7&f:4;&=ESC&_x:1&f:3;&=F1&=F2&=F3&=F4&=F5&=F6&_x:0.5;&=INS&=HOME&=PAGE%20UP&=PRINT&_x:0.5;&=PAUSE&_x:2;&=NUM%20LOCK&_f:6;&=//&=*&=/&ndash/;;&@_x:2&f:3;&=F7&=F8&=F9&=F10&=F11&=F12&_x:0.5;&=DEL&=END&=PAGE%20DOWN&_f:2;&=SCROLL%20LOCK&_x:3.5&f:8;&=7&=8&=9&_f:6&h:2;&=+;&@_x:16&f:8;&=4&_n:true;&=5&=6;&@_a:5&f:5;&=~%0A%60&=!%0A1&=/@%0A2&=#%0A3&=$%0A4&=%25%0A5&=%5E%0A6&=/&%0A7&=*%0A8&=(%0A9&=)%0A0&=/&mdash/;%0A/&ndash/;&=+%0A/=&_a:7&f:3&w:2;&=BACK%20SPACE&_x:1&f:8;&=1&=2&=3&_f:3&h:2;&=ENTER;&@_w:1.5;&=TAB&_f:8;&=Q&=W&=E&=R&=T&=Y&=U&=I&=O&=P&_a:5&f:5;&=%7B%0A%5B&=%7D%0A%5D&_w:1.5;&=%7C%0A%5C&_x:1&a:7&f:8&w:2;&=0&=.;&@_f:3&w:1.25&w2:1.75&l:true;&=CAPS%20LOCK&_x:0.5&f:8;&=A&=S&=D&_n:true;&=F&=G&=H&_n:true;&=J&=K&=L&_a:5&f:5;&=/:%0A/;&=%22%0A'&_a:7&f:3&w:2.25;&=ENTER;&@_w:2.25;&=SHIFT&_f:8;&=Z&=X&=C&=V&=B&=N&=M&_a:5&f:5;&=%3C%0A,&=%3E%0A.&=?%0A//&_a:7&f:3&w:2;&=SHIFT;&@_y:-0.75&x:14.5&f:6;&=/&#9652/;;&@_y:-0.25&f:3&w:1.5;&=CTRL&_w:1.5;&=ALT&_p=DSA%20SPACE&w:7;&=&_p=DSA&w:1.5;&=ALT&_w:1.5;&=CTRL;&@_y:-0.75&x:13.5&f:6;&=/&#9666/;&=/&#9662/;&=/&#9656/), the raw data is include costar orientation. 
QMK firmware is amazing! I like the extra features. But I want to try to keep it as original as possible on the external interface. In this case, the line sequence and costar is same as the the original pcb, so we can make the costar assembly the same, and select to use the origin daughterboard or not. If not, we can modify the keycode mapping in the firmware to be compatible.

**Comment by vuckale** [comment]
@Curious-r I initially wanted to do the same but original matrix was too clustered for me and decided to do it from scratch. But respect to you for trying to keep it as original as possible!

Here is a quick screenshot of the daughterboard I designed for it:
![Screenshot_2024-01-22-23-23-06_3440x1440](https://github.com/vuckale/compaq_mx_11800_qmk_via/assets/56733276/4192294d-c569-4f70-8d15-feb5b85576a6)

Let me know when you finish your board and if you decide to sell some of your PCBs I would gladly buy one!

**Comment by Curious-r** [comment]
@vuckale Sorry for the late reply, you know, life..
I'm almost done, but I'm half a world away from you, and it should be inconvenient to express the PCB. I would be happy to send you the gerber file by email and you can make a PCB proof anywhere. 😁
No need to buy, please just leave me an email address.

**Comment by vuckale** [comment]
@Curious-r Hello again, sorry it took me so long to reply! I would be happy if you send me the gerbers when you have the PCB ready! Please send it here: **tiler_apropos.0w @ icloud.com**

I have setup this temporal e-mail to prevent spam. Just remove spaces from the address I pasted above and write a comment again here when you send me so that I know. Thank you! 😁

**Comment by Curious-r** [comment]
@vuckale Done, check the mailbox! 
Note that we must install the diode on bottom and solder it on the same side, after soldering we should cut the pins on top side. In this way, the diode won't prevent us from unplugging the switch. If you don't need diode, just replacement it with wire.
If you have any questions, just contact me. 

Enjoy! 😊

**Comment by Curious-r** [comment]
@vuckale Sorry, I made a mistake the first time, the right stuff is named with "Gerber_G80_11801.zip".

**Comment by vuckale** [comment]
@Curious-r Hey I have got it thank you. There is a small problem... I am from European Union and we use ISO layout here so my keycaps are not going to fit your PCB. Is it possible you can send me the design files so I can adjust the layout? Are you using kiCAD to develop PCB?

**Comment by Curious-r** [comment]
@vuckale I'm using lceda instead of kiCAD, it is a software only in chinese area, this is the reason I didn't send the schematic diagram before.. Now I send it to you, maybe you can try to edit it with EasyEDA, but I'm not really sure it can work well. Or I can export it to AD.

**Comment by vuckale** [comment]
@Curious-r I will see what I can do and will let you know. Thank you very much! 😊
